app/src/main/java/com/cornellappdev/android/eatery/presenter/WeeklyPresenter.java

```diff
@@ -14,64 +14,66 @@
+    private View view;
+    private ArrayList<EateryBaseModel> mEateryList;
+    private LocalDate mSelectedDate;
+    private Repository rInstance = Repository.getInstance();
 
-  public WeeklyPresenter(View view) {
-    this.view = view;
-    mEateryList = rInstance.getEateryList();
-    mSelectedDate = LocalDate.now();
-  }
+    public WeeklyPresenter(View view) {
+        this.view = view;
+        mEateryList = rInstance.getEateryList();
+        mSelectedDate = LocalDate.now();
+    }
 
-  public LocalDate getDayInWeek(int offset) {
-    return LocalDate.now().plusDays(offset);
-  }
+    public LocalDate getDayInWeek(int offset) {
+        return LocalDate.now().plusDays(offset);
+    }
 
-  public LocalDate getSelectedDate() {
-    return mSelectedDate;
-  }
+    public LocalDate getSelectedDate() {
+        return mSelectedDate;
+    }
 
-  public void setSelectedDate(LocalDate selectedDate) {
-    mSelectedDate = selectedDate;
-  }
+    public void setSelectedDate(LocalDate selectedDate) {
+        mSelectedDate = selectedDate;
+    }
 
-  public ArrayList<DiningHallModel> getDiningHallList() {
-    ArrayList<DiningHallModel> diningHallList = new ArrayList<>();
-    for (EateryBaseModel m : mEateryList) {
-      if (m instanceof DiningHallModel) {
-        diningHallList.add((DiningHallModel) m);
-      }
+    public ArrayList<DiningHallModel> getDiningHallList() {
+        ArrayList<DiningHallModel> diningHallList = new ArrayList<>();
+        for (EateryBaseModel m : mEateryList) {
+            if (m instanceof DiningHallModel) {
+                diningHallList.add((DiningHallModel) m);
+            }
+        }
+        return diningHallList;
     }
-    return diningHallList;
-  }
 
-  public HashMap<String, ArrayList<DiningHallModel>> generateAreaLists(MealType mealType, LocalDate date) {
-    ArrayList<DiningHallModel> westList = new ArrayList<>();
-    ArrayList<DiningHallModel> northList = new ArrayList<>();
-    ArrayList<DiningHallModel> centralList = new ArrayList<>();
-    for (DiningHallModel dhm : getDiningHallList()) {
-      if (dhm.getMealByDateAndType(date, mealType) != null ||
-          (mealType == MealType.LUNCH && dhm.getMealByDateAndType(date, MealType.BRUNCH) != null)) {
-        switch (dhm.getArea()) {
-          case NORTH:
-            northList.add(dhm);
-            break;
-          case WEST:
-            westList.add(dhm);
-            break;
-          case CENTRAL:
-            centralList.add(dhm);
-            break;
+    public HashMap<String, ArrayList<DiningHallModel>> generateAreaLists(MealType mealType,
+            LocalDate date) {
+        ArrayList<DiningHallModel> westList = new ArrayList<>();
+        ArrayList<DiningHallModel> northList = new ArrayList<>();
+        ArrayList<DiningHallModel> centralList = new ArrayList<>();
+        for (DiningHallModel dhm : getDiningHallList()) {
+            if (dhm.getMealByDateAndType(date, mealType) != null ||
```

Comment:
can we split the two conditions, write them to boolean values and give them descriptive names so that the code is self-documenting